src/Element.h

```diff
@@ -10,6 +10,7 @@
 #define epanet_rtx_element_h
 
 #include <string.h>
+#include <unordered_map>
```

Comment:
Well, as the names(keys) as you mentioned cannot be repeated, unordered_map will give us O(1) time complexity for inserting , removing and adding new metadata into the map, while std::map also makes sure that the keys are always sorted via BST, and that means that these functionalities are gonna cost us O(log(n)) time

Reply:
ok, but let's keep in mind that these metadata maps are going to be small `n` - only a couple of values for specific elements. additionally, we will likely want the keys to be sorted for consistency across APIs. seems more convenient to use `map` since the complexity tradeoff is not significant.